src/txn/txn.c

```diff
@@ -1126,11 +1130,14 @@ __txn_resolve_prepared_op(WT_SESSION_IMPL *session, WT_TXN_OP *op, bool commit,
          */
         if (upd->txnid == WT_TXN_ABORTED)
             continue;
-        if (upd->txnid != txn->id)
+        if (upd->txnid != txn->id) {
+            WT_ASSERT(session, (upd->next == NULL || upd->next->txnid != txn->id));
```

Comment:
no need for extra brackets around `upd->next == NULL || upd->next->txnid != txn->id`

Reply:
I am interested in checking for all the remaining updates for `upd->txnid != txn->id`. We might have couple of updates from another transaction for example, followed by updates from the this transaction. My suggestion is to loop through rest of updates in the chain here. This will only be done with diagnostic compile flag.

Reply:
I reused the loop that you added when exiting from the resolve function.